app.js

```diff
@@ -207,6 +217,7 @@ webmakerProxy(app, middleware.checkForAuth);
 // Main page
 app.get('/',
         middleware.setNewPageOperation,
+        middleware.decryptToken,
```

Comment:
I'll rename it `maybeDecryptToken`, unless you can think of a better name. If it finds a token, it decrypts it and populates `req.user`. Otherwise it just calls `next()`.